### PR TITLE
[v1.26] Allow downgrading from v1.26+ to v1.25.x

### PR DESCRIPTION
**Pull request by rene-dekker**
For Calico Cloud it is not an uncommon scenario that a v1.26+ cluster is turned into a v1.25.x cluster. Some standardization and refactoring are making it hard to downgrade, causing a bad experience. This PR addresses these compatibility issues.

### CODE REVIEW

File: pkg/controller/installation/core_controller.go [hunk 1]
```diff
@@ -1436,11 +1441,28 @@ func (r *ReconcileInstallation) GetTyphaNodeTLSConfig() (*render.TyphaNodeTLS, e
...
 	}
 
+	if ca == nil && typha != nil && typha.Data[corev1.TLSCertKey] != nil && node != nil && node.Data[corev1.TLSCertKey] != nil {
+		// This is a cluster downgraded from v1.26+, where the configmap is no longer automatically created.
+		// We can create a compatible configmap by combining the PEM data of both certificates.
+		ca = &corev1.ConfigMap{
+			TypeMeta:   metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: render.TyphaCAConfigMapName, Namespace: common.OperatorNamespace()},
+			Data: map[string]string{
+				render.TyphaCABundleName: fmt.Sprintf("# certificate: node-certs\n%s\n\n# certificate: node-certs\n%s", string(typha.Data[corev1.TLSCertKey]), string(node.Data[corev1.TLSCertKey])),
```
Comment (tmjd) [inline pkg/controller/installation/core_controller.go:1461]: This is what I meant earlier. Both say node-certs, I think the 1st was intended to say typha-certs.
I didn't realize a bundle was literally the contents of multiple certs in the same file.

Comment (rene-dekker) [reply pkg/controller/installation/core_controller.go:1461]: I'll rectify the typo.

If you follow the docs you would likely only put the CA PEM in here, but placing the two individual certificate PEMs in here works as well as an alternative.

**Review by tmjd — APPROVED** [review]
You've mentioned this works for you and it looks good AFAICT. I think my concern I mentioned in slack is unrelated to this specific change. It also looks like you've guarded the changes enough that this should not affect anything unexpectedly.
tldr; this LGTM :grinning:
